custom_components/aseko_local/aseko_decoder.py

```diff
@@ -56,10 +80,47 @@ def _unit_type(
         _LOGGER.warning(error)
         raise ValueError(error)
 
+        # # 1. Serial number must be valid (first 4 bytes, big endian)
```

Comment:
Seems as something, that might be covered in tests in case it is not yet.

And then I would remove this long comments section from here.